core/templates/dev/head/components/forms/schema_editors/SchemaBasedListEditorDirective.js

```diff
@@ -135,12 +135,23 @@ oppia.directive('schemaBasedListEditor', [
             }
           };
 
+          var _deleteElementIfEmpty = function() {
+            for (var i = 0; i < $scope.localValue.length - 1; i++) {
+              if ($scope.localValue[i].length === 0) {
+                $scope.deleteElement(i);
+                i--;
+              }
+            }
+          };
+
           $scope.lastElementOnBlur = function() {
             _deleteLastElementIfUndefined();
+            _deleteElementIfEmpty();
```

Comment:
In schema_based_list_editor_directive we have
on-input-blur="($last ? lastElementOnBlur : showAddItemButton)" so whenever the input is blurred the function is checking if that field is empty and deletes it if it is empty by calling _deleteElementIfEmpty.

Ahh, I get it now we need to pass the index in my previous logic all the fields were being checked but I guess that too does not create any problem.